src/vellum/workflows/nodes/displayable/bases/search_node.py

```diff
@@ -73,11 +74,13 @@ class Outputs(BaseOutputs):
         results: List[SearchResult]
 
     def _perform_search(self) -> SearchResponse:
+        index_is_uuid = True if isinstance(self.document_index, UUID) else is_valid_uuid(self.document_index)
+
         try:
             return self._context.vellum_client.search(
                 query=self.query,
-                index_id=str(self.document_index) if isinstance(self.document_index, UUID) else None,
-                index_name=self.document_index if isinstance(self.document_index, str) else None,
+                index_id=str(self.document_index) if index_is_uuid else None,
+                index_name=str(self.document_index) if not index_is_uuid else None,
```

Comment:
[oos] this is where the `document_index` param could be useful as we'd be able to just do `document_index=str(self.document_index)` without the isinstance/is_valid_uuid logic above